### DISCUSSION TITLE
Get fastest IP in background

### DISCUSSION THREAD

**Discussion by fonlan** [AdguardTeam/AdGuardHome#3671] [Ideas]
Currently if enable "Fastest IP address" function it will slow down the query speed. 

My suggestion is when the query domain was not in cache, query it from upstream and return all without speedtest.
After this domain save to cache, ADH can use some background thread to test the fastest IP and modify the cache. Only keep the fastest one.
Next time when this query is return from cache ADH can return the fastest one directly.

This solution can bring "Fastest IP addresss" function without additional query delay.

**Comment by BK8000L** [comment]
bump. if the message above wasn't clear: first time AGH should return the fastest response (just like parallel requests mode, but with 10sec ttl), and then AGH refreshes the response in background to the fastest ip.
This will elliminate slow downs on "fastest ip" mode.
